app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,88 @@
+class TweetsController < ApplicationController
+
+  get "/tweets" do
+    if logged_in?
+      @user = User.find_by(username: session[:username])
+      erb :"/tweets/index"
+    else
+      redirect "/login"
+    end
+  end
+
+
+  get '/tweets/new' do
+    if logged_in?
+      @user = User.find_by(params[:user])
+      erb :"/tweets/new"
+    else
+      redirect :'/login'
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :"/tweets/show"
+    else
+      redirect :'/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      if @tweet && @tweet.user.username == session[:username]
+        erb :"/tweets/edit"
+      else
+        redirect '/tweets'
+      end
+    else
+      redirect :'/login'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content]==""
+      redirect "/tweets/new"
+    else
+      @tweet = Tweet.create(content: params[:content])
+      @tweet.user = User.find_by(username: session[:username])
+      @tweet.save
+      redirect "tweets/#{@tweet.id}"
+    end
+  end
+
+  patch '/tweets/:id' do
+    if logged_in?
+      if params[:content] == ""
+        redirect to "/tweets/#{params[:id]}/edit"
+      else
+        @tweet = Tweet.find_by_id(params[:id])
+        if @tweet && @tweet.user.username == session[:username]
+          if @tweet.update(content: params[:content])
+            redirect to "/tweets/#{@tweet.id}"
+          else
+            redirect to "/tweets/#{@tweet.id}/edit"
+          end
+        else
+          redirect to '/tweets'
+        end
+      end
+    else
+      redirect to '/login'
+    end
+  end
+
+  delete '/tweets/:id/delete' do
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.